In [1]:
import pandas as pd
import numpy as np

In [292]:
movie_industry = pd.read_csv("../data/movie_industry.csv", encoding = "ISO-8859-1" )
movie_industry.head()

budget                                 company country        director  \
0   8000000.0           Columbia Pictures Corporation     USA      Rob Reiner   
1   6000000.0                      Paramount Pictures     USA     John Hughes   
2  15000000.0                      Paramount Pictures     USA      Tony Scott   
3  18500000.0  Twentieth Century Fox Film Corporation     USA   James Cameron   
4   9000000.0                    Walt Disney Pictures     USA  Randal Kleiser   

       genre        gross                      name rating    released  \
0  Adventure   52287414.0               Stand by Me      R  1986-08-22   
1     Comedy   70136369.0  Ferris Bueller's Day Off  PG-13  1986-06-11   
2     Action  179800601.0                   Top Gun     PG  1986-05-16   
3     Action   85160248.0                    Aliens      R  1986-07-18   
4  Adventure   18564613.0   Flight of the Navigator     PG  1986-08-01   

   runtime  score               star   votes         writer  year  
0       89    8.1        Wil Wheaton  299174   Stephen King  1986  
1      103    7.8  Matthew Broderick  264740    John Hughes  1986  
2      110    6.9         Tom Cruise  236909       Jim Cash  1986  
3      137    8.4   Sigourney Weaver  540152  James Cameron  1986  
4       90    6.9        Joey Cramer   36636  Mark H. Baker  1986

In [293]:
sum(movie_industry.gross.isna())

0

In [294]:
sum(movie_industry.gross == 0)

0

In [295]:
movie_industry.groupby('year').count()

budget  company  country  director  genre  gross  name  rating  \
year                                                                   
1986     220      220      220       220    220    220   220     220   
1987     219      219      219       219    219    219   219     219   
1988     220      220      220       220    220    220   220     220   
1989     221      221      221       221    221    221   221     221   
1990     220      220      220       220    220    220   220     220   
1991     220      220      220       220    220    220   220     220   
1992     220      220      220       220    220    220   220     220   
1993     220      220      220       220    220    220   220     220   
1994     220      220      220       220    220    220   220     220   
1995     220      220      220       220    220    220   220     220   
1996     220      220      220       220    220    220   220     220   
1997     220      220      220       220    220    220   220     220   
1998     220      220      220       220    220    220   220     220   
1999     220      220      220       220    220    220   220     220   
2000     220      220      220       220    220    220   220     220   
2001     220      220      220       220    220    220   220     220   
2002     220      220      220       220    220    220   220     220   
2003     220      220      220       220    220    220   220     220   
2004     220      220      220       220    220    220   220     220   
2005     220      220      220       220    220    220   220     220   
2006     220      220      220       220    220    220   220     220   
2007     220      220      220       220    220    220   220     220   
2008     220      220      220       220    220    220   220     220   
2009     220      220      220       220    220    220   220     220   
2010     220      220      220       220    220    220   220     220   
2011     220      220      220       220    220    220   220     220   
2012     220      220      220       220    220    220   220     220   
2013     220      220      220       220    220    220   220     220   
2014     220      220      220       220    220    220   220     220   
2015     220      220      220       220    220    220   220     220   
2016     220      220      220       220    220    220   220     220   

      released  runtime  score  star  votes  writer  
year                                                 
1986       220      220    220   220    220     220  
1987       219      219    219   219    219     219  
1988       220      220    220   220    220     220  
1989       221      221    221   221    221     221  
1990       220      220    220   220    220     220  
1991       220      220    220   220    220     220  
1992       220      220    220   220    220     220  
1993       220      220    220   220    220     220  
1994       220      220    220   220    220     220  
1995       220      220    220   220    220     220  
1996       220      220    220   220    220     220  
1997       220      220    220   220    220     220  
1998       220      220    220   220    220     220  
1999       220      220    220   220    220     220  
2000       220      220    220   220    220     220  
2001       220      220    220   220    220     220  
2002       220      220    220   220    220     220  
2003       220      220    220   220    220     220  
2004       220      220    220   220    220     220  
2005       220      220    220   220    220     220  
2006       220      220    220   220    220     220  
2007       220      220    220   220    220     220  
2008       220      220    220   220    220     220  
2009       220      220    220   220    220     220  
2010       220      220    220   220    220     220  
2011       220      220    220   220    220     220  
2012       220      220    220   220    220     220  
2013       220      220    220   220    220     220  
2014     

In [296]:
movie_industry[movie_industry.name.str.contains("Nemo")]

budget                  company country        director      genre  \
728   35000000.0        TMS Entertainment   Japan     Masami Hata     Family   
3750  94000000.0  Pixar Animation Studios     USA  Andrew Stanton  Animation   

            gross                                    name rating    released  \
728     1368210.0  Little Nemo: Adventures in Slumberland      G  1992-08-21   
3750  380843261.0                            Finding Nemo      G  2003-05-30   

      runtime  score           star   votes          writer  year  
728        85    7.2  Gabriel Damon    5800  Chris Columbus  1989  
3750      100    8.1  Albert Brooks  772029  Andrew Stanton  2003

In [297]:
years = np.unique(movie_industry.year.values)
years

array([1986, 1987, 1988, 1989, 1990, 1991, 1992, 1993, 1994, 1995, 1996,
       1997, 1998, 1999, 2000, 2001, 2002, 2003, 2004, 2005, 2006, 2007,
       2008, 2009, 2010, 2011, 2012, 2013, 2014, 2015, 2016])

In [298]:
directors = np.unique(movie_industry.director.values)
directors

array(['A.R. Murugadoss', 'Aamir Khan', 'Aaron Blaise', ...,
       'Éric Rohmer', 'Étienne Chatiliez', 'Éva Gárdos'], dtype=object)

In [299]:
actors = np.unique(movie_industry.star.values)
actors

array(["'Weird Al' Yankovic", '50 Cent', 'A.J. Cook', ..., 'Álex Angulo',
       'Élodie Bouchez', 'Émilie Dequenne'], dtype=object)

In [300]:
individuals = np.append(directors, actors)
individuals

array(['A.R. Murugadoss', 'Aamir Khan', 'Aaron Blaise', ...,
       'Álex Angulo', 'Élodie Bouchez', 'Émilie Dequenne'], dtype=object)

In [301]:
years2 = np.repeat(years, len(individuals))
years2

array([1986, 1986, 1986, ..., 2016, 2016, 2016])

In [302]:
agents = np.tile(individuals, len(years))
agents

array(['A.R. Murugadoss', 'Aamir Khan', 'Aaron Blaise', ...,
       'Álex Angulo', 'Élodie Bouchez', 'Émilie Dequenne'], dtype=object)

In [303]:
is_director = np.tile(np.append(np.ones(len(directors)), np.zeros(len(actors))), len(years))
is_director

array([1., 1., 1., ..., 0., 0., 0.])

In [304]:
ID = np.tile(np.arange(len(individuals)), len(years))
ID

array([   0,    1,    2, ..., 5260, 5261, 5262])

In [305]:
time = np.repeat(np.arange(len(years)), len(individuals))
time

array([ 0,  0,  0, ..., 30, 30, 30])

In [306]:
assert len(agents) == len(years2)
assert len(agents) == len(is_director)
assert len(agents) == len(ID)

In [307]:
df  = pd.DataFrame({"agent": agents,
                   "is_director": is_director,
                   "id": ID,
                   "year": years2,
                   "time": time})
df

agent  is_director    id  year  time
0          A.R. Murugadoss          1.0     0  1986     0
1               Aamir Khan          1.0     1  1986     0
2             Aaron Blaise          1.0     2  1986     0
3             Aaron Harvey          1.0     3  1986     0
4       Aaron Kim Johnston          1.0     4  1986     0
...                    ...          ...   ...   ...   ...
163148         Zoey Deutch          0.0  5258  2016    30
163149     Zooey Deschanel          0.0  5259  2016    30
163150         Álex Angulo          0.0  5260  2016    30
163151      Élodie Bouchez          0.0  5261  2016    30
163152     Émilie Dequenne          0.0  5262  2016    30

[163153 rows x 5 columns]

In [308]:
assert df.groupby("is_director").count().agent[1] == len(directors)*len(years)
assert len(df.groupby(["agent", "is_director"]).count()) == len(individuals)

In [309]:
len(individuals)  - len(df.groupby(["agent"]).count())

171

There are 171 people who double and actors and directors

In [310]:
df[df.agent == "Clint Eastwood"]

agent  is_director    id  year  time
459     Clint Eastwood          1.0   459  1986     0
3201    Clint Eastwood          0.0  3201  1986     0
5722    Clint Eastwood          1.0   459  1987     1
8464    Clint Eastwood          0.0  3201  1987     1
10985   Clint Eastwood          1.0   459  1988     2
...                ...          ...   ...   ...   ...
150565  Clint Eastwood          0.0  3201  2014    28
153086  Clint Eastwood          1.0   459  2015    29
155828  Clint Eastwood          0.0  3201  2015    29
158349  Clint Eastwood          1.0   459  2016    30
161091  Clint Eastwood          0.0  3201  2016    30

[62 rows x 5 columns]

In [311]:
%%time
total_gross = []
mean_score = []
mean_votes = []
for i in df.index:
    if df.is_director[i]:
        temp = movie_industry[movie_industry.director == df.agent[i]]
        temp = temp[temp.year <= df.year[i]]
        if len(temp) > 0 :  
            gross = sum(temp.gross.values)
            score = np.mean(temp.score.values)
            votes = np.mean(temp.votes.values)
        else:
            gross = 0
            score = 0
            votes = 0
        total_gross.append(gross)
        mean_score.append(score)
        mean_votes.append(votes)
    else:
        temp = movie_industry[movie_industry.star == df.agent[i]]
        temp = temp[temp.year <= df.year[i]]
        if len(temp) > 0 :  
            gross = sum(temp.gross.values)
            score = np.mean(temp.score.values)
            votes = np.mean(temp.votes.values)
        else:
            gross = 0
            score = 0
            votes = 0
        total_gross.append(gross)
        mean_score.append(score)
        mean_votes.append(votes)

CPU times: user 4min 57s, sys: 2.58 s, total: 5min
Wall time: 5min 19s


In [312]:
df["log_total_gross"] = [np.log(i) if i != 0 else i for i in total_gross]
df["mean_score"] = mean_score
df["mean_votes"] = mean_votes

In [313]:
df

agent  is_director    id  year  time  log_total_gross  \
0          A.R. Murugadoss          1.0     0  1986     0         0.000000   
1               Aamir Khan          1.0     1  1986     0         0.000000   
2             Aaron Blaise          1.0     2  1986     0         0.000000   
3             Aaron Harvey          1.0     3  1986     0         0.000000   
4       Aaron Kim Johnston          1.0     4  1986     0         0.000000   
...                    ...          ...   ...   ...   ...              ...   
163148         Zoey Deutch          0.0  5258  2016    30        18.035911   
163149     Zooey Deschanel          0.0  5259  2016    30        17.310201   
163150         Álex Angulo          0.0  5260  2016    30         9.637176   
163151      Élodie Bouchez          0.0  5261  2016    30        13.729860   
163152     Émilie Dequenne          0.0  5262  2016    30        12.474094   

        mean_score  mean_votes  
0             0.00         0.0  
1             0.00         0.0  
2             0.00         0.0  
3             0.00         0.0  
4             0.00         0.0  
...            ...         ...  
163148        5.95     54405.5  
163149        7.30    208674.5  
163150        7.50     13867.0  
163151        7.60      4397.0  
163152        7.60     10176.0  

[163153 rows x 8 columns]

In [314]:
genre = np.unique(movie_industry.genre.values)
genre

array(['Action', 'Adventure', 'Animation', 'Biography', 'Comedy', 'Crime',
       'Drama', 'Family', 'Fantasy', 'Horror', 'Musical', 'Mystery',
       'Romance', 'Sci-Fi', 'Thriller', 'War', 'Western'], dtype=object)

In [320]:
%%time
d = {}
for gen in genre:
    d[gen] = []   
for i in df.index:
    if df.is_director[i]:
        temp = movie_industry[movie_industry.director == df.agent[i]]
    else:
        temp = movie_industry[movie_industry.star == df.agent[i]]
    temp = temp[temp.year <= df.year[i]]
    if len(temp) > 0:
        temp = temp.groupby(["genre"])["director"].count()
        for gen in genre:
            if gen in temp.index:
                d[gen].append(temp[gen])
            else:
                d[gen].append(0)
    else:
        for gen in genre:
            d[gen].append(0)

CPU times: user 5min 47s, sys: 2.82 s, total: 5min 50s
Wall time: 6min 3s


In [323]:
temp = pd.DataFrame(d)
df = df.merge(temp, left_index =True, right_index =True)
df

agent  is_director    id  year  time  log_total_gross  \
0          A.R. Murugadoss          1.0     0  1986     0         0.000000   
1               Aamir Khan          1.0     1  1986     0         0.000000   
2             Aaron Blaise          1.0     2  1986     0         0.000000   
3             Aaron Harvey          1.0     3  1986     0         0.000000   
4       Aaron Kim Johnston          1.0     4  1986     0         0.000000   
...                    ...          ...   ...   ...   ...              ...   
163148         Zoey Deutch          0.0  5258  2016    30        18.035911   
163149     Zooey Deschanel          0.0  5259  2016    30        17.310201   
163150         Álex Angulo          0.0  5260  2016    30         9.637176   
163151      Élodie Bouchez          0.0  5261  2016    30        13.729860   
163152     Émilie Dequenne          0.0  5262  2016    30        12.474094   

        mean_score  mean_votes  Action  Adventure  ...  Family  Fantasy  \
0             0.00         0.0       0          0  ...       0        0   
1             0.00         0.0       0          0  ...       0        0   
2             0.00         0.0       0          0  ...       0        0   
3             0.00         0.0       0          0  ...       0        0   
4             0.00         0.0       0          0  ...       0        0   
...            ...         ...     ...        ...  ...     ...      ...   
163148        5.95     54405.5       1          0  ...       0        0   
163149        7.30    208674.5       0          0  ...       0        0   
163150        7.50     13867.0       1          0  ...       0        0   
163151        7.60      4397.0       0          0  ...       0        0   
163152        7.60     10176.0       0          0  ...       0        0   

        Horror  Musical  Mystery  Romance  Sci-Fi  Thriller  War  Western  
0            0        0        0        0       0         0    0        0  
1            0        0        0        0       0         0    0        0  
2            0        0        0        0       0         0    0        0  
3            0        0        0        0       0         0    0        0  
4            0        0        0        0       0         0    0        0  
...        ...      ...      ...      ...     ...       ...  ...      ...  
163148       0        0        0        0       0         0    0        0  
163149       0        0        0        0       0         0    0        0  
163150       0        0        0        0       0         0    0        0  
163151       0        0        0        0       0         0    0        0  
163152       0        0        0        0       0         0    0        0  

[163153 rows x 25 columns]

In [324]:
ratings = np.unique(movie_industry.rating.values)
ratings

array(['B', 'B15', 'G', 'NC-17', 'NOT RATED', 'Not specified', 'PG',
       'PG-13', 'R', 'TV-14', 'TV-MA', 'TV-PG', 'UNRATED'], dtype=object)

In [326]:
%%time
d = {}
for gen in ratings:
    d[gen] = []   
for i in df.index:
    if df.is_director[i]:
        temp = movie_industry[movie_industry.director == df.agent[i]]
    else:
        temp = movie_industry[movie_industry.star == df.agent[i]]
    temp = temp[temp.year <= df.year[i]]
    if len(temp) > 0:
        temp = temp.groupby(["rating"])["director"].count()
        for gen in ratings:
            if gen in temp.index:
                d[gen].append(temp[gen])
            else:
                d[gen].append(0)
    else:
        for gen in ratings:
            d[gen].append(0)

CPU times: user 5min 47s, sys: 2.82 s, total: 5min 50s
Wall time: 6min 3s


In [330]:
temp = pd.DataFrame(d)
df = df.merge(temp, left_index =True, right_index =True)
df

agent  is_director    id  year  time  log_total_gross  \
0          A.R. Murugadoss          1.0     0  1986     0         0.000000   
1               Aamir Khan          1.0     1  1986     0         0.000000   
2             Aaron Blaise          1.0     2  1986     0         0.000000   
3             Aaron Harvey          1.0     3  1986     0         0.000000   
4       Aaron Kim Johnston          1.0     4  1986     0         0.000000   
...                    ...          ...   ...   ...   ...              ...   
163148         Zoey Deutch          0.0  5258  2016    30        18.035911   
163149     Zooey Deschanel          0.0  5259  2016    30        17.310201   
163150         Álex Angulo          0.0  5260  2016    30         9.637176   
163151      Élodie Bouchez          0.0  5261  2016    30        13.729860   
163152     Émilie Dequenne          0.0  5262  2016    30        12.474094   

        mean_score  mean_votes  Action  Adventure  ...  NC-17  NOT RATED  \
0             0.00         0.0       0          0  ...      0          0   
1             0.00         0.0       0          0  ...      0          0   
2             0.00         0.0       0          0  ...      0          0   
3             0.00         0.0       0          0  ...      0          0   
4             0.00         0.0       0          0  ...      0          0   
...            ...         ...     ...        ...  ...    ...        ...   
163148        5.95     54405.5       1          0  ...      0          0   
163149        7.30    208674.5       0          0  ...      0          0   
163150        7.50     13867.0       1          0  ...      0          0   
163151        7.60      4397.0       0          0  ...      0          1   
163152        7.60     10176.0       0          0  ...      0          0   

        Not specified  PG  PG-13  R  TV-14  TV-MA  TV-PG  UNRATED  
0                   0   0      0  0      0      0      0        0  
1                   0   0      0  0      0      0      0        0  
2                   0   0      0  0      0      0      0        0  
3                   0   0      0  0      0      0      0        0  
4                   0   0      0  0      0      0      0        0  
...               ...  ..    ... ..    ...    ...    ...      ...  
163148              0   0      1  1      0      0      0        0  
163149              0   0      1  1      0      0      0        0  
163150              0   0      0  1      0      0      0        0  
163151              0   0      0  0      0      0      0        0  
163152              0   0      0  1      0      0      0        0  

[163153 rows x 38 columns]

In [339]:
oscar = pd.read_csv("../data/the_oscar_award.csv")
oscar.category = oscar.category.str.lower()
filtered = oscar[oscar.category.str.contains("(actor)|(actress)|(directing)", regex = True)]
filtered.loc[filtered.category.str.contains("(actor)|(actress)"), 'is_director'] = 0
filtered.loc[filtered.category.str.contains("directing"), 'is_director'] = 1

In [340]:
oscar_ppl = filtered[~filtered.name.str.contains("( and)|(,)|(and )", regex = True)]
oscar_ppl.is_director = oscar_ppl.is_director.astype(np.uint8)
oscar_ppl.head()

year_film  year_ceremony  ceremony category                 name  \
0       1927           1928         1    actor  Richard Barthelmess   
1       1927           1928         1    actor        Emil Jannings   
2       1927           1928         1  actress       Louise Dresser   
3       1927           1928         1  actress         Janet Gaynor   
4       1927           1928         1  actress       Gloria Swanson   

               film  winner  is_director  
0         The Noose   False            0  
1  The Last Command    True            0  
2   A Ship Comes In   False            0  
3        7th Heaven    True            0  
4    Sadie Thompson   False            0

In [341]:
oscar_ppl.shape

(2192, 8)

In [342]:
filtered[filtered.name.str.contains("( and)|(,)|(and )", regex = True)]

year_film  year_ceremony  ceremony                    category  \
531        1937           1938        10  actor in a supporting role   
3768       1961           1962        34                   directing   
5655       1978           1979        51                   directing   
5942       1981           1982        54  actor in a supporting role   
6043       1982           1983        55  actor in a supporting role   
6289       1984           1985        57                   directing   
6506       1986           1987        59                   directing   
7558       1996           1997        69  actor in a supporting role   
8832       2007           2008        80                   directing   
9184       2010           2011        83                   directing   

                             name                        film  winner  \
531                  Roland Young                      Topper   False   
3768  Robert Wise, Jerome Robbins             West Side Story    True   
5655    Warren Beatty, Buck Henry             Heaven Can Wait   False   
5942       Howard E. Rollins, Jr.                     Ragtime   False   
6043           Louis Gossett, Jr.  An Officer and a Gentleman    True   
6289                 Roland Joffé          The Killing Fields   False   
6506                 Roland Joffé                 The Mission   False   
7558            Cuba Gooding, Jr.               Jerry Maguire    True   
8832     Joel Coen and Ethan Coen      No Country for Old Men    True   
9184     Joel Coen and Ethan Coen                   True Grit   False   

      is_director  
531           0.0  
3768          1.0  
5655          1.0  
5942          0.0  
6043          0.0  
6289          1.0  
6506          1.0  
7558          0.0  
8832          1.0  
9184          1.0

In [343]:
#Manually modify the rows where there are special characters (for movie after 1986)
oscar_ppl = pd.concat([oscar_ppl, filtered[(filtered.index==7558) | (filtered.index==6506)]])
oscar_ppl.shape

(2194, 8)

In [344]:
oscar_ppl[oscar_ppl.name=="Joel Coen and Ethan Coen"]

Empty DataFrame
Columns: [year_film, year_ceremony, ceremony, category, name, film, winner, is_director]
Index: []

In [345]:
oscar_ppl = pd.concat([oscar_ppl, filtered[(filtered.index==8832) | (filtered.index==9184)]], ignore_index=False)
oscar_ppl = pd.concat([oscar_ppl, filtered[(filtered.index==8832) | (filtered.index==9184)]], ignore_index=True)
oscar_ppl.shape

(2198, 8)

In [346]:
oscar_ppl[oscar_ppl.name== "Joel Coen and Ethan Coen"]

year_film  year_ceremony  ceremony   category                      name  \
2194       2007           2008        80  directing  Joel Coen and Ethan Coen   
2195       2010           2011        83  directing  Joel Coen and Ethan Coen   
2196       2007           2008        80  directing  Joel Coen and Ethan Coen   
2197       2010           2011        83  directing  Joel Coen and Ethan Coen   

                        film  winner  is_director  
2194  No Country for Old Men    True          1.0  
2195               True Grit   False          1.0  
2196  No Country for Old Men    True          1.0  
2197               True Grit   False          1.0

In [347]:
oscar_ppl.at[2194, 'name'] = 'Joel Coen'
oscar_ppl.at[2195, 'name'] = 'Ethan Coen'
oscar_ppl.at[2196, 'name'] = 'Ethan Coen'
oscar_ppl.at[2197, 'name'] = 'Joel Coen'

In [348]:
oscar_ppl.shape

(2198, 8)

In [349]:
oscar_ppl = oscar_ppl[oscar_ppl['year_film']>1985]
print('oscar award dataset has shape {}'.format(oscar_ppl.shape))
oscar_ppl.head()

oscar award dataset has shape (852, 8)


year_film  year_ceremony  ceremony                 category  \
1346       1986           1987        59  actor in a leading role   
1347       1986           1987        59  actor in a leading role   
1348       1986           1987        59  actor in a leading role   
1349       1986           1987        59  actor in a leading role   
1350       1986           1987        59  actor in a leading role   

               name                      film  winner  is_director  
1346  Dexter Gordon           'Round Midnight   False          0.0  
1347    Bob Hoskins                 Mona Lisa   False          0.0  
1348   William Hurt  Children of a Lesser God   False          0.0  
1349    Paul Newman        The Color of Money    True          0.0  
1350    James Woods                  Salvador   False          0.0

In [350]:
%%time
total_nominations =[]
total_win_counts=[]
for i in df.index:
    if df.is_director[i]:
        temp = oscar_ppl[(oscar_ppl.name == df.agent[i]) & (oscar_ppl.is_director == True)]
        temp = temp[temp.year_film <= df.year[i]]
        nomination = temp.shape[0]
        total_nominations.append(nomination)
        win = sum(temp.winner.values)
        total_win_counts.append(win)
    else:
        temp = oscar_ppl[(oscar_ppl.name == df.agent[i]) & (oscar_ppl.is_director == False)]
        temp = temp[temp.year_film <= df.year[i]]
        nomination = temp.shape[0]
        total_nominations.append(nomination)
        win = sum(temp.winner.values)
        total_win_counts.append(win)

CPU times: user 5min 45s, sys: 1.56 s, total: 5min 46s
Wall time: 5min 57s


In [351]:
df["nominations"] = total_nominations
df["win_counts"] = total_win_counts
df.head()

agent  is_director  id  year  time  log_total_gross  \
0     A.R. Murugadoss          1.0   0  1986     0              0.0   
1          Aamir Khan          1.0   1  1986     0              0.0   
2        Aaron Blaise          1.0   2  1986     0              0.0   
3        Aaron Harvey          1.0   3  1986     0              0.0   
4  Aaron Kim Johnston          1.0   4  1986     0              0.0   

   mean_score  mean_votes  Action  Adventure  ...  Not specified  PG  PG-13  \
0         0.0         0.0       0          0  ...              0   0      0   
1         0.0         0.0       0          0  ...              0   0      0   
2         0.0         0.0       0          0  ...              0   0      0   
3         0.0         0.0       0          0  ...              0   0      0   
4         0.0         0.0       0          0  ...              0   0      0   

   R  TV-14  TV-MA  TV-PG  UNRATED  nominations  win_counts  
0  0      0      0      0        0            0           0  
1  0      0      0      0        0            0           0  
2  0      0      0      0        0            0           0  
3  0      0      0      0        0            0           0  
4  0      0      0      0        0            0           0  

[5 rows x 40 columns]

In [352]:
df

agent  is_director    id  year  time  log_total_gross  \
0          A.R. Murugadoss          1.0     0  1986     0         0.000000   
1               Aamir Khan          1.0     1  1986     0         0.000000   
2             Aaron Blaise          1.0     2  1986     0         0.000000   
3             Aaron Harvey          1.0     3  1986     0         0.000000   
4       Aaron Kim Johnston          1.0     4  1986     0         0.000000   
...                    ...          ...   ...   ...   ...              ...   
163148         Zoey Deutch          0.0  5258  2016    30        18.035911   
163149     Zooey Deschanel          0.0  5259  2016    30        17.310201   
163150         Álex Angulo          0.0  5260  2016    30         9.637176   
163151      Élodie Bouchez          0.0  5261  2016    30        13.729860   
163152     Émilie Dequenne          0.0  5262  2016    30        12.474094   

        mean_score  mean_votes  Action  Adventure  ...  Not specified  PG  \
0             0.00         0.0       0          0  ...              0   0   
1             0.00         0.0       0          0  ...              0   0   
2             0.00         0.0       0          0  ...              0   0   
3             0.00         0.0       0          0  ...              0   0   
4             0.00         0.0       0          0  ...              0   0   
...            ...         ...     ...        ...  ...            ...  ..   
163148        5.95     54405.5       1          0  ...              0   0   
163149        7.30    208674.5       0          0  ...              0   0   
163150        7.50     13867.0       1          0  ...              0   0   
163151        7.60      4397.0       0          0  ...              0   0   
163152        7.60     10176.0       0          0  ...              0   0   

        PG-13  R  TV-14  TV-MA  TV-PG  UNRATED  nominations  win_counts  
0           0  0      0      0      0        0            0           0  
1           0  0      0      0      0        0            0           0  
2           0  0      0      0      0        0            0           0  
3           0  0      0      0      0        0            0           0  
4           0  0      0      0      0        0            0           0  
...       ... ..    ...    ...    ...      ...          ...         ...  
163148      1  1      0      0      0        0            0           0  
163149      1  1      0      0      0        0            0           0  
163150      0  1      0      0      0        0            0           0  
163151      0  0      0      0      0        0            0           0  
163152      0  1      0      0      0        0            0           0  

[163153 rows x 40 columns]

In [353]:
df.to_csv("../data/features.csv", index= False)